spec/generators/install_generator_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe HtmxRails::Generators::InstallGenerator, type: :generator do
+  destination File.expand_path('../tmp', __dir__)
+
+  let!(:generate_files) {}
+
+  before(:each) do
+    prepare_destination
+    generate_files
+  end
+
+  after(:all) { FileUtils.rm_rf destination_root }
+
+  def generate_application_js(location_folder)
+    pathname = File.join([destination_root, location_folder])
+    FileUtils.mkdir_p(pathname)
+    File.write("#{pathname}/application.js", '')
+  end
+
+  context 'with Sprockets' do
+    before(:each) do
+      run_generator
+    end
+
+    context 'when `application.js` exists' do
+      let!(:generate_files) do
```

Comment:
this looks more like a `before` block than with lets 🤔

Reply:
I tried to make a kind of 'hook'. I call `generate_files` [here](https://github.com/rootstrap/htmx-rails/blob/install-generators/spec/generators/install_generator_spec.rb#L12), so I can create the files, but also following the order I need (`prepare_destination`, `[generate_files]`, `run_generator`) and avoid to repeat the `before` block within each context.

I wasn't very sure about this, but I am running out of ideas 😂

Reply:
I think I'd do the following:
1. global before block (for all the tests)
```ruby
  before do
    prepare_destination
  end
```
2. local before block
```ruby
before do
  generate_files
end
```
3. it block
```ruby
it '...' do
  run_generator
  # ...
end
```
or in any case, moving the `run_generator` right below the `generate_files` (or on their own `before` block)

if doing that it'd be more clear the order of execution